fontforgeexe/fontview.c

```diff
@@ -1029,25 +1029,191 @@ char *GetPostScriptFontName(char *dir, int mult) {
 return( temp );
 }
 
+struct gfc_data {   //GFileChooser
+    int done;
+    int ret;
+    GGadget *gfc;
+    GGadget *ignore_invalid;
+    SplineFont *sf;
+    EncMap *map;
+};
+
+static int feat_e_h(GWindow gw, GEvent *event) {
+    if ( event->type==et_close ) {
+    struct gfc_data *d = GDrawGetUserData(gw);
+    d->done = true;
+    d->ret = false;
+    } else if ( event->type==et_char ) {
+return( false );
+    } else if ( event->type==et_map ) {
+    GDrawRaise(gw);
+    } else if ( event->type==et_mousemove ||
+        (event->type==et_mousedown && event->u.mouse.button==3 )) {
+    struct gfc_data *d = GDrawGetUserData(gw);
+    GFileChooserPopupCheck(d->gfc,event);
+    } else if (( event->type==et_mouseup || event->type==et_mousedown ) &&
+        (event->u.mouse.button>=4 && event->u.mouse.button<=7) ) {
+    struct gfc_data *d = GDrawGetUserData(gw);
+return( GGadgetDispatchEvent((GGadget *) (d->gfc),event));
+    }
+return( true );
+}
+
+static int FEAT_ImportOk(GGadget *g, GEvent *e) {
+    if ( e->type==et_controlevent && e->u.control.subtype==et_buttonactivate ) {
+        struct gfc_data *d = GDrawGetUserData(GGadgetGetWindow(g));
+        unichar_t *ret = GGadgetGetTitle(d->gfc);
+        char *temp = u2def_copy(ret);
+        int ignore_invalid = GGadgetIsChecked(d->ignore_invalid);
+        GGadget *tf;
+
+        GFileChooserGetChildren(d->gfc,NULL,NULL,&tf);
+        if ( *_GGadgetGetTitle(tf)=='\0' )
+            return( true );
+        GDrawSetCursor(GGadgetGetWindow(g),ct_watch);
+
+        if ( !LoadKerningDataFromMetricsFile(d->sf,temp,d->map,ignore_invalid))
+            ff_post_error(_("Load of Kerning Metrics Failed"),_("Failed to load kern data from %s"), temp);
+        d->done = true;
+
+        GDrawSetCursor(GGadgetGetWindow(g),ct_pointer);
+        free(ret); free(temp);
+    }
+    return( true );
+
+}
+
+static int FEAT_Cancel(GGadget *g, GEvent *e) {
+    if ( e->type==et_controlevent && e->u.control.subtype == et_buttonactivate ) {
+        struct gfc_data *d = GDrawGetUserData(GGadgetGetWindow(g));
+        d->done = true;
+        d->ret = false;
+    }
+    return( true );
+}
+
 void MergeKernInfo(SplineFont *sf,EncMap *map) {
 #ifndef __Mac
-    static char wild[] = "*.{afm,tfm,ofm,pfm,bin,hqx,dfont,feature,feat,fea}";
-    static char wild2[] = "*.{afm,amfm,tfm,ofm,pfm,bin,hqx,dfont,feature,feat,fea}";
+    static unichar_t wild[] = {'*','.','{','a','f','m',',','t','f','m',',','o','f','m',',','p','f','m',',','b','i','n',',','h','q','x',',','d','f','o','n','t',',','f','e','a','t','u','r','e',',','f','e','a','t',',','f','e','a','}'};
```

Comment:
I don't have time to check this at the moment, but all of this needs a closer look. It's a big code smell to be converting what was a normal char to unichar_t. None of these literals are null terminated.

Reply:
It's because there's no "8" version of the function he now needs. I feel it's too much to ask him to implement one.

Reply:
GFileChooserSetFilterText8 doesn't exist, so he needs GFileChooserSetFilterText

See also my comment here: https://github.com/fontforge/fontforge/pull/4097#discussion_r373361576

If we agree to allow C11 we can fix this, and many other constants, tomorrow ;-)

Reply:
I'm not convinced, it's almost never a good idea to be working directly with unichars, unless you're working deep in gdraw.

Reply:
He is working deep in gdraw :man_shrugging: 

I've needed the same function before.

Reply:
(As to whether they should be null-terminated, I think you are probably right and they should be, which is a good catch :-) )

Reply:
Well if it's needed I don't think it's that hard to add. I'd much rather that than have to do this.

I don't think we should be enforcing C11 yet either.

Reply:
I am trying not to bite a new contributor. I already felt I was making a tall order when I asked him to implement this in the GUI as well.

However, I take your point @jtanx. I will personally implement GFileChooserSetFilterText8 for him, as I've already asked a lot of him. Would that be acceptable to you?

Reply:
ok

Reply:
It's a security bug missing null termination; thanks for spotting it. Fixed using the new API now.